src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -351,29 +361,78 @@ public static bool IsOnlyWrittenTo(this ExpressionSyntax expression)
                         return true;
                     }
                 }
+
+                if (IsExpressionOfArgumentInDeconstruction(expression))
+                {
+                    return true;
+                }
             }
 
             return false;
         }
 
-        public static bool IsAttributeNamedArgumentIdentifier(this ExpressionSyntax expression)
+        /// <summary>
+        /// If this declaration or identifier is part of a deconstruction, find the deconstruction.
+        /// If found, returns either an assignment expression or a foreach variable statement.
+        /// Returns null otherwise.
+        /// 
+        /// copied from SyntaxExtensions.GetContainingDeconstruction
+        /// </summary>
+        private static bool IsExpressionOfArgumentInDeconstruction(ExpressionSyntax expr)
```

Comment:
talked to @jcouv and he says this is hte right way to determine if you're on the assignment part of a deconstruction.

Reply:
There is a semantic model API has a `GetDeconstructionInfo`. Looking at it, it seems I didn't add compiler tests (filed https://github.com/dotnet/roslyn/issues/26352 just now) :-S
I think it will only return something if you pass in an assignment or foreach syntax nodes. 
That means you could indeed just check that you're on the left of an assignment (unwrapping tuples) and that the assignment is a deconstruction (using `GetDeconstructionInfo`).

I'm not sure that buys much (you still need to walk up the node's ancestors to find an assignment or foreach), and that's doing some binding work, so I'd recommend sticking with a purely syntactic check, as currently implemented in the PR.

Reply:
@jcouv How does this appear through IOperatoin?  is is possible it's better?  or would we still be lacking the info that this is a write and not a read?